cfme/tests/control/test_actions.py

```diff
@@ -697,3 +723,28 @@ def finalize():
     clone_request = Request(description=request_description, partial_check=True)
     clone_request.wait_for_request(method='ui')
     assert clone_request.status == "Error"
+
+
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(VMwareProvider, RHEVMProvider,
+    OpenStackProvider, AzureProvider))
+def test_action_check_compliance(request, provider, vm, vm_name, policy_for_testing):
+    """Tests action "Check Host or VM Compliance". Policy profile should have control and compliance
+    policies. Control policy initiates compliance check and compliance policy determines is the vm
+    compliant or not. After reloading vm details screen the compliance status should be changed.
+    """
+    if any(tag.category.display_name == "Service Level" and tag.display_name == "Gold"
+           for tag in vm.crud.get_tags()):
+        vm.crud.remove_tag(("Service Level", "Gold"))
+
+    @request.addfinalizer
+    def _remove_tag():
+        if any(tag.category.display_name == "Service Level" and tag.display_name == "Gold"
+               for tag in vm.crud.get_tags()):
+            vm.crud.remove_tag(("Service Level", "Gold"))
+
+    policy_for_testing.assign_actions_to_event("Tag Complete", ["Check Host or VM Compliance"])
+    request.addfinalizer(policy_for_testing.assign_events)
+    vm.crud.add_tag(("Service Level", "Gold"), single_value=True)
+    view = navigate_to(vm.crud, "Details")
+    view.toolbar.reload.click()
+    assert vm.crud.compliant
```

Comment:
**QUESTION:** Would asserting vm.crud.compliant not require the two previous lines if it refreshed the page before checking?

Reply:
@psav It's true, but the current [implementation](https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/vm.py#L209) doesn't allow that. It will require `BaseVM` refactoring to fix that.

Reply:
Can we get a TODO in there that his needs to be altered please?

Reply:
then I'm happy to merge - you can cite this function as a model